### ISSUE TITLE
Avoid copyup on touch

### ISSUE THREAD

**Issue by kcgthb** [stanford-rc/fuse-migratefs#18]
Treat `open(..., O_FOLLOW)` as a special case to avoid migrating files that are `touch`ed.

Make this an settable option, like #8 and #9

**Comment by giuseppe** [comment]
if it can be helpful, containers/fuse-overlayfs implemented copyup using reflinks (where the underlying file system supports it), so only the inode is created on touch.

**Comment by thiell** [comment]
@giuseppe that's a cool feature, but reflinks only work when you are on the same filesystem right, and also only with btrfs? In our case, lower and upper are different filesystems.

**Comment by giuseppe** [comment]
Yes correct, also xfs supports reflinks now.  fuse-overlayfs doesn't enforce it but the lower and upper layer are supposed to be on the same file system.  Otherwise how do you enforce that the inodes are different?

**Comment by synnack** [comment]
@giuseppe overlayfs for migrations or for read/write overlay on read only filesystem most certainly never have the lower and upper on the same file system. That's basically the whole point of migratefs.
